### PR TITLE
[App Service] `az logicapp create`, `az appservice plan create` : Fix #20337 : add support for App Service Plan Logic App SKUs (WS1-3) and auto-creation of plans when creating a Logic App 

### LINKED ISSUE #20337: can't create logic app with standard plan as I can't create the service app plan workflow based.
as in description. 

I try to create a logic app standard plan (app service plan for workflow) but there is no option to create an app service plan with workflow.

### PR DESCRIPTION
**Pull request by StrawnSC**
**Description**

Allows the creation of App Service Plans with SKUs WS1, WS2, or WS3. These SKUs are used for plans that contain Logic Apps. This fixes #20337

Also changes the behavior of `az logicapp create`. If neither `--plan` nor `--consumption-plan-location` are specified, instead of failing, the command creates a WS1 App Service Plan to house the Logic App 

**Testing Guide**

`azdev test --live test_logicapp_commands`

**History Notes**

[App Service] `az logicapp create`: Autogenerate a WS1 App Service Plan when no value for `--plan` or `--consumption-plan-location` is provided 
[App Service] `az appservice plan create`: Allow creating App Service Plans for Logic Apps (SKUs WS1, WS2, and WS3) 

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [X] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [X] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [X] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

File: src/azure-cli/azure/cli/command_modules/appservice/custom.py [hunk 1]
```diff
@@ -1822,6 +1822,9 @@ def create_app_service_plan(cmd, resource_group_name, name, is_linux, hyper_v, p
                               reserved=(is_linux or None), hyper_v=(hyper_v or None), name=name,
                               per_site_scaling=per_site_scaling, hosting_environment_profile=ase_def)
 
+    if sku.upper() in ['WS1', 'WS2', 'WS3']:
```
Comment (panchagnula) [inline src/azure-cli/azure/cli/command_modules/appservice/custom.py:1825]: @StrawnSC given CLI commands are supposed to idempotent, what is the experience if an existing webapp/functionapp tries to use the LogicApp specific SKUS? Would be good to add some exceptional handling existing non logic apps Serverfarms to try to change to the new SKUs added. Please check with LA team if there is some property on ASPs to identify as LogicApp specific?

**Review by panchagnula — CHANGES_REQUESTED** [review]
Added a question.

File: src/azure-cli/azure/cli/command_modules/appservice/custom.py (hunk 1 above)
Comment (StrawnSC) [reply src/azure-cli/azure/cli/command_modules/appservice/custom.py:1825]: Yikes it succeeded in updating the SKU to WS1 when I tried the command on a normal ASP that already exists. I'll ask about how to distinguish logic app ASPs
Thanks for pointing that out, I hadn't considered a user trying to update an ASP with the command

Comment (panchagnula) [reply src/azure-cli/azure/cli/command_modules/appservice/custom.py:1825]: Ouch - please file a bugs on Management loop (API) to handle not updating in this case?

Comment (StrawnSC) [reply src/azure-cli/azure/cli/command_modules/appservice/custom.py:1825]: I put a bug report on the control plane ADO and added the client-side validation in the CLI 
According to @archidda the best way to tell if an app service plan is a logic app plan is to just check if `plan.sku.tier == 'WorkflowStandard'`

**Comment by StrawnSC** [comment]
@zhoxing-ms one of the extension loading CI tests here is failing with: 
`ERROR: Request to https://github.com/Azure/azure-hanaonazure-cli-extension/releases/download/0.6.5/sap_hana-0.6.5-py2.py3-none-any.whl failed with 403` 
Do you know what's causing this?

**Comment by zhoxing-ms** [comment]
@StrawnSC This is a issue related to other extensions. Could you address those conflicts for this PR first?

**Review by panchagnula — APPROVED** [review]
LGTM

**Review by zhoxing-ms — APPROVED** [review]

**Comment by Poonamn91** [comment]
Hi, Do we have a solution for this issue? i am unable to create a Logic app with sku WS1, there is no argument to pass either sku or a app service plan name. Please suggest

**Comment by StrawnSC** [comment]
Sorry you're running into an issue here, @Poonamn91. 
When I run `az appservice plan create -n <plan_name> -g <group_name> --sku WS1` (filling in `<plan_name>` and `<group_name>` with the desired name and resource group name), it successfully creates a WS1 app service plan.

Please ensure you're using the latest version of the azure CLI. You can upgrade your CLI version with `az upgrade`. When  you run `az --version`, you should see the CLI version shown as `2.34.1`

**Comment by Poonamn91** [comment]
@StrawnSC  thanks for the reply. We are also trying to assign the app service plan to the logic app. But it's not applying the standard sku , it is creating with basic sku. Please suggest the command for logic app to pass the sku ws1.

**Comment by Poonamn91** [comment]
Hi Team, can someone please suggest a command to create a logic app with Standard Sku, not able to find a argument to pass sku.

**Comment by panchagnula** [comment]
@Poonamn91 as per this doc for logic app standard the following are the supported SKUs https://docs.microsoft.com/en-us/azure/logic-apps/logic-apps-pricing#standard-pricing  (Ws1, ws2, ws3) & these are supported using CLI as well.

**Pull request merged**